kong/plugins/moesif/regex_config_helpers.lua

```diff
@@ -0,0 +1,180 @@
+        local conditions = regex_rule["conditions"]
+        -- Bool flag to determine if the regex conditions are matched
+        local regex_matched = nil 
+        -- Create a table to hold the conditions mapping (path and value)
+        local condition_table = {}
+
+        -- Iterate through the regex rule conditions and map the path and value
+        for _, condition in pairs(conditions) do
+            -- Check if the path does not exists already in the table (when multiple path like request.route exist in the conditions) 
+            if condition_table[condition["path"]] ~= nil then 
+                -- Create a table to hold the existing values
+                local append_value = {}
+                -- Check if the path if of type table
+                if type(condition_table[condition["path"]]) == "table" then
+                    -- Iterate through the path values and append value 
+                    for _, value in ipairs(condition_table[condition["path"]]) do 
+                        table.insert(append_value, value)
+                    end
+                else 
+                    -- Append existing value to the table
+                    table.insert(append_value, condition_table[condition["path"]])
+                end
+                -- Append new value to the existing values
+                table.insert(append_value, condition["value"])
+                -- Add condition path -> values (new and existing value) to the condition table
+                condition_table[condition["path"]] = append_value
+            else
+                -- Add condition path -> value to the condition table
+                condition_table[condition["path"]] = condition["value"]
+            end
+        end
+
+        -- Iterate through conditions table and perform `and` operation between each conditions
+        for path, values in pairs(condition_table) do 
+            -- Check if the path exists in the request config mapping
+            if request_config_mapping[path] ~= nil then 
+                -- Fetch the value of the path in request config mapping
+                local event_data = request_config_mapping[path]
+                -- Check if the condition values is of type table and not nil
+                if type(values) == 'table' and next(values) ~= nil then 
+                    -- Iterate through all the values and perform regex matching with event value
+                    for _, value in pairs(values) do 
+                        regex_matched = regex_match(event_data, value, regex_matched)
+                    end 
+                else 
+                    -- Perform regex matching with event value
+                    regex_matched = regex_match(event_data, values, regex_matched)
+                end         
+            else 
+                -- Path does not exists, so regex condition not matched, performing `and` operation with previous condition
+                regex_matched = (regex_matched and false)
```

Comment:
this is trickier..and i think to be safe here.. for not..in case when there is no available path, just break from this loop and assume regex rule not matched